src/clipboard.js

```diff
@@ -26,21 +29,36 @@ export function copyNode(node: Element) {
   selection.removeAllRanges()
 }
 
-export function copyText(text: string) {
+export function copyText(button: Element, text: string) {
+  if (writeAsync(button, text)) return
+
   const body = document.body
   if (!body) return
 
   const node = createNode(text)
   body.appendChild(node)
-  copyNode(node)
+  copyNode(button, node)
   body.removeChild(node)
 }
 
-export function copyInput(node: HTMLInputElement | HTMLTextAreaElement) {
+export function copyInput(button: Element, node: HTMLInputElement | HTMLTextAreaElement) {
+  if (writeAsync(button, node.value)) return
+
   node.select()
   document.execCommand('copy')
   const selection = getSelection()
   if (selection != null) {
     selection.removeAllRanges()
   }
 }
+
+function writeAsync(button: Element, text: string): boolean {
+  // $FlowFixMe Clipboard is not defined in Flow yet.
```

Comment:
gotta get that flow type contribution opened hehe

Reply:
https://github.com/facebook/flow/pull/6324